feat(curriculum): add workshop python functions (#68279)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-kitchen-inventory-tracker/6a3e6fd08125dfa615cb431b.md

---
id: 6a3e6fd08125dfa615cb431b
title: Step 10
challengeType: 20
dashedName: step-10
---

# --description--

> Now that your `use_eggs` function is complete, it is time to use it. When you call a function that requires **parameters**, you have to provide values for them inside the parentheses. These values are called **arguments**.

Reviewer: ```suggestion
Now that your `use_eggs` function is complete, you can test it out. When calling a function that requires **parameters**, you pass the specific values inside the parentheses. These values are known as **arguments**.
```

> At the bottom of your code, call your `use_eggs` function and pass it two arguments: the global `available_eggs` variable and the number `1` (representing the number of eggs you want to use).

Reviewer: Let's add an example on how to call function with arguments here.

> At the bottom of your code, call your `use_eggs` function and pass it two arguments: the global `available_eggs` variable, and the number `1` (representing the number of eggs you want to use).

Reviewer: ```suggestion
At the bottom of your code, call your `use_eggs` function and pass it two arguments: the global `available_eggs` variable and the number `1` (representing the number of eggs you want to use).
```

# --hints--

You should call the `use_eggs` function and pass `eggs` and `1` as arguments.

```js
({
  test: () => assert(
    runPython(`_Node(_code).has_call("use_eggs(eggs, 1)")`)
  )
})
```

# --seed--

## --seed-contents--

```py
eggs = 1
flour = 2
sugar = 3

def take_inventory():
    total_items = eggs + flour + sugar
    print(f'The kitchen has {total_items} total items:')
    print(f'- {eggs} eggs')
    print(f'- {flour} flour')
    print(f'- {sugar} sugar')

take_inventory()

def use_eggs(eggs, num_eggs):
    if num_eggs > eggs:
        print('The kitchen does not have enough eggs.')
        return eggs

    print(f'{num_eggs} egg(s) used out of {eggs} available.')
    return eggs - num_eggs
        

> --fcc-editable-region--
>         
> --fcc-editable-region--

Reviewer 2: ```suggestion
--fcc-editable-region--

--fcc-editable-region--
```

---
id: 6a3e6fd08125dfa615cb431b
title: Step 10
challengeType: 20
dashedName: step-10
---

# --description--

Now that your `use_eggs` function is complete, you can test it out. When calling a function that requires **parameters**, you pass the specific values inside the parentheses. These values are known as **arguments**.

At the bottom of your code, call your `use_eggs` function and pass it two arguments: the global `available_eggs` variable and the number `1` (representing the number of eggs you want to use).

Here is an example of how to call a function with arguments:

```py
def calculate_sum(a, b):
    print(a + b)
calculate_sum(3, 1) # Output: 4

> ```    

Reviewer: This should come after the first paragraph.
